From 0962f87bab8a0c637321ff727d3285b8a627d0a8 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Mon, 5 Jun 2023 22:59:00 +0200
Subject: [PATCH 63/95] arm64: dts: st: add dsi support on stm32mp251

This patch adds support for MIPI-DSI interface on stm32mp251.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git e1487828de850a8d207cf484b2c91076b249bfc5 arm64: dts: st: add dsi support on stm32mp251]
Signed-off-by: Philippe Cornu <philippe.cornu@foss.st.com>
Signed-off-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Change-Id: I8d5effde9655c762304be5c90985e1c5991939d
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 6a08816acd9e3..088e05ecbeb76 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -1584,6 +1584,19 @@ trigger@4 {
 				};
 			};
 
+			dsi: dsi@48000000 {
+				compatible = "st,stm32-dsi";
+				reg = <0x48000000 0x800>;
+				#clock-cells = <0>;
+				clocks = <&rcc CK_BUS_DSI>, <&rcc CK_KER_DSIPHY>,
+					 <&rcc CK_KER_LTDC>;
+				clock-names = "pclk", "ref", "px_clk";
+				resets = <&rcc DSI_R>;
+				reset-names = "apb";
+				access-controllers = <&rifsc 81>;
+				status = "disabled";
+			};
+
 			ltdc: display-controller@48010000 {
 				compatible = "st,stm32mp251-ltdc";
 				reg = <0x48010000 0x400>;
-- 
2.51.0

